package/MDAnalysis/analysis/helix_analysis.py

```diff
@@ -418,49 +420,49 @@ def _prepare(self):
         for key, dims in self.attr_shapes.items():
             empty = [self._zeros_per_frame(
                 dims, n_positions=n) for n in n_res]
-            setattr(self, key, empty)
+            setattr(self.results, key, empty)
```

Comment:
fyi: Could now be written as
```suggestion
            self.results[key] = empty
```